### PR TITLE
Added support for ramp-pricing subscriptions

### PR DESCRIPTION
**Pull request by csampson**
- Adds `ramp_intervals` attribute to `Recurly::Subscription`
- Adds `Recurly::SubscriptionRampInterval` class, which can be used to initialize ramp_intervals for subscriptions via the `ramp_intervals` attribute

### CODE REVIEW

File: spec/fixtures/subscriptions/serialize-with-ramps.xml [hunk 1]
```diff
@@ -0,0 +1 @@
+<subscription><account><account_code>1</account_code><billing_info><month>1</month><number>4111-1111-1111-1111</number><year>2014</year></billing_info><dunning_campaign_id>1234abcd</dunning_campaign_id><email>verena@example.com</email><first_name>Verena</first_name><last_name>Example</last_name></account><currency>EUR</currency><customer_notes>Some Customer Notes</customer_notes><imported_trial>true</imported_trial><plan_code>gold</plan_code><ramp_intervals><ramp_interval><starting_billing_cycle>1</starting_billing_cycle><unit_amount_in_cents>1000</unit_amount_in_cents></ramp_interval><ramp_interval><starting_billing_cycle>2</starting_billing_cycle><unit_amount_in_cents>2000</unit_amount_in_cents></ramp_interval></ramp_intervals><shipping_address_id>1234</shipping_address_id><shipping_amount_in_cents>899</shipping_amount_in_cents><shipping_method_code>ups_ground</shipping_method_code><terms_and_conditions>Some Terms and Conditions</terms_and_conditions></subscription>
```
Comment (Patrick-Duvall) [inline spec/fixtures/subscriptions/serialize-with-ramps.xml:1]: Is it worth formatting this?

File: spec/recurly/subscription_spec.rb [hunk 2]
```diff
@@ -69,6 +69,28 @@
...
+              starting_billing_cycle: 2,
+              unit_amount_in_cents: 2000
+            )
+          ]
+        )
+      end
+
+      it 'properly serializes' do
+        subscription = Subscription.new attributes
+        subscription.to_xml.must_equal get_raw_xml('subscriptions/serialize-with-ramps.xml')
```
Comment (Patrick-Duvall) [inline spec/recurly/subscription_spec.rb:91]: Now I see why its formatted like that.

**Review by Patrick-Duvall — APPROVED** [review]
Given the approach we took for plans and the fact most stuff i specced in resource.rb I feel good with this.

File: spec/fixtures/subscriptions/serialize-with-ramps.xml (hunk 1 above)
Comment (mwlang) [reply spec/fixtures/subscriptions/serialize-with-ramps.xml:1]: yes!
```xml
<subscription>
  <account>
    <account_code>1</account_code>
    <billing_info>
      <month>1</month>
      <number>4111-1111-1111-1111</number>
      <year>2014</year>
    </billing_info>
    <dunning_campaign_id>1234abcd</dunning_campaign_id>
    <email>verena@example.com</email>
    <first_name>Verena</first_name>
    <last_name>Example</last_name>
  </account>
  <currency>EUR</currency>
  <customer_notes>Some Customer Notes</customer_notes>
  <imported_trial>true</imported_trial>
  <plan_code>gold</plan_code>
  <ramp_intervals>
    <ramp_interval>
      <starting_billing_cycle>1</starting_billing_cycle>
      <unit_amount_in_cents>1000</unit_amount_in_cents>
    </ramp_interval>
    <ramp_interval>
      <starting_billing_cycle>2</starting_billing_cycle>
      <unit_amount_in_cents>2000</unit_amount_in_cents>
    </ramp_interval>
  </ramp_intervals>
  <shipping_address_id>1234</shipping_address_id>
  <shipping_amount_in_cents>899</shipping_amount_in_cents>
  <shipping_method_code>ups_ground</shipping_method_code>
  <terms_and_conditions>Some Terms and Conditions</terms_and_conditions>
</subscription>
```

File: spec/recurly/subscription_spec.rb (hunk 2 above)
Comment (mwlang) [reply spec/recurly/subscription_spec.rb:91]:
```ruby
subscription.to_xml.must_equal get_raw_xml('subscriptions/serialize-with-ramps.xml').split("\n").map(&:strip).join
```

if we're doing this sort of equality checks regularly, then perhaps add a 2nd optional parameter to get_raw_xml for `compact: true` and encapsulate the `.split("\n").map(&:strip).join` bit.

**Pull request merged**